### PR TITLE
feat(vite-plugin): support different extensions for catalogs. Improve coverage

### PR DESCRIPTION
**Pull request by thekip**
# Description

Part of custom formatters ongoing work. Prepare loaders to works with not only `*.po`.

[//]: # (Describe the big picture of your changes here to communicate to the maintainers why we should accept this pull request. If it fixes a bug or resolves a feature request, be sure to link to that issue.)

## Types of changes

[//]: # (What types of changes does your code introduce to Lingui?)
[//]: # (_Put an `x` in the boxes that apply_)

- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation update

Fixes # (issue)

## Checklist

[//]: # (_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code._)

- [ ] I have read the [CONTRIBUTING](https://github.com/lingui/js-lingui/blob/main/CONTRIBUTING.md) and [CODE_OF_CONDUCT](https://github.com/lingui/js-lingui/blob/main/CODE_OF_CONDUCT.md) docs
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)

### CODE REVIEW

**Comment by thekip** [comment]
Blocked by build pipeline. Release should be build before running this test. I will look for solutions

**Comment by thekip** [comment]
@andrii-bodnar please stop windows WFs here. Looks like they have no time limit and already executing for couple hours 😮

**Comment by andrii-bodnar** [comment]
@thekip stopped. It was running for 3 hours 😬

I will look into time limiting it tomottow

File: packages/vite-plugin/README.md [hunk 1]
```diff
@@ -4,7 +4,7 @@
 
 # @lingui/vite-plugin
 
-> Vite plugin which compiles on the fly the .po files for auto-refreshing. In summary, `lingui compile` command isn't required when using this plugin
+> Vite plugin which compiles on the fly lingui catalogs. In summary, `lingui compile` command isn't required when using this plugin
```
Comment (andrii-bodnar) [inline packages/vite-plugin/README.md:7]:
```suggestion
> Vite plugin that compiles Lingui catalogs on the fly. In summary, the `lingui compile` command isn't needed when using this plugin.
```

File: website/docs/ref/vite-plugin.md [hunk 2]
```diff
@@ -38,6 +38,14 @@ export async function dynamicActivate(locale: string) {
 }
 ```
 
+:::note
+If you use format which has different extension then `*.po` you need to specify `?lingui` suffix
```
Comment (andrii-bodnar) [inline website/docs/ref/vite-plugin.md:42]:
```suggestion
If you are using a format that has a different extension than `*.po`, you need to specify the `?lingui` suffix:
```

**Comment by usercao** [comment]
Pleasant update, praising your work.

**Pull request merged**